controllers/healthcheck_controller.go

```diff
@@ -152,6 +153,10 @@ func (r *HealthCheckReconciler) processOrRecoverHealthCheck(ctx context.Context,
 	// Process HealthCheck
 	ret, procErr := r.processHealthCheck(ctx, log, healthCheck)
 	if procErr != nil {
+		// do retry without error during "the object has been modified; please apply your changes to the latest version and try again"
+		if r.IsOptimisticLockError(procErr) {
```

Comment:
This is a conflict error, client-go has a utility to `retry.RetryOnConflict` which would work for this case.
https://pkg.go.dev/k8s.io/client-go/util/retry#RetryOnConflict